src/plugins/messageStorage/text.js

```diff
@@ -46,24 +46,67 @@ class TextFileMessageStorage {
 		const time = moment(msg.time).utcOffset(tz).format(format);
 		let line = `[${time}] `;
 
-		if (msg.type === "message") {
-			// Format:
-			// [2014-01-01 00:00:00] <Arnold> Put that cookie down.. Now!!
-			line += `<${msg.from.nick}> ${msg.text}`;
-		} else {
-			// Format:
-			// [2014-01-01 00:00:00] * Arnold quit
-			line += `* ${msg.from.nick} `;
+		switch (msg.type) {
+		case "message":
+			// [2014-01-01 00:00:00] <@Arnold> Put that cookie down.. Now!!
+			line += `<${msg.from.mode}${msg.from.nick}> ${msg.text}`;
+			break;
+		case "action":
+			// [2014-01-01 00:00:00] * @Arnold is eating cookies
+			line += `* ${msg.from.mode}${msg.from.nick} ${msg.text}`;
+			break;
+		case "notice":
+			// [2014-01-01 00:00:00] -Arnold- pssst... I have cookies!
+			line += `-${msg.from.nick}- ${msg.text}`;
+			break;
+		case "join":
+			// [2014-01-01 00:00:00] *** Joins: Arnold (~arnold@foo.bar)
+			line += `*** Joins: ${msg.from.nick} (${msg.hostmask})`;
+			break;
+		case "part":
+			// [2014-01-01 00:00:00] *** Parts: Arnold (~arnold@foo.bar) (Bye all!)
+			line += `*** Parts: ${msg.from.nick} (${msg.hostmask}) (${msg.text})`;
+			break;
+		case "quit":
+			// [2014-01-01 00:00:00] *** Quits: Arnold (~arnold@foo.bar) (Connection reset by peer)
+			line += `*** Quits: ${msg.from.nick} (${msg.hostmask}) (${msg.text})`;
+			break;
+		case "nick":
+			// [2014-01-01 00:00:00] *** Arnold is now known as Bernie
+			line += `*** ${msg.from.nick} is now known as ${msg.new_nick}`;
+			break;
+		case "kick":
+			// [2014-01-01 00:00:00] *** Arnold was kicked by Bernie (Don't steal my cookies!)
+			line += `*** ${msg.target.nick} was kicked by ${msg.from.nick} (${msg.text})`;
+			break;
+		case "mode":
+			// [2014-01-01 00:00:00] *** Arnold sets mode: +o Bernie
+			line += `*** ${msg.from.nick} sets mode: ${msg.text}`;
+			break;
+		case "topic":
+			// ignore topic lines you get on join
+			if (typeof msg.from.nick === "undefined") {
+				return;
+			}
+
+			// [2014-01-01 00:00:00] *** Arnold changes topic to 'welcome everyone!'
+			line += `*** ${msg.from.nick} changes topic to '${msg.text}'`;
```

Comment:
I recommend "changes topic to:" instead to avoid

(1) trailing or leading whitespace being painfully obvious
(2) quotes in the text itself not looking misleading

Reply:
This is following ZNC's format, as we're trying to follow it, and not invent our own.